### ISSUE TITLE
Encrypted Client Hello

### ISSUE THREAD

**Issue by nhorman** [openssl/project#659]
A significant amount of work has been done on Encrypted Client Hello by a community member, and we should work toward getting it integrated into openssl.

Source material I believe is here:
https://github.com/sftcd/openssl/tree/ECH-draft-13c

**Comment by sftcd** [comment]
Hiya, for scheduling purposes, it'd be great to know when this is likely to start up. I assume @nhorman the kick-off is you create the feature branch, then tell me what you'd like as a 1st PR for that, I go make that PR and then we iterate? FWIW, I'm happy to start on that anytime, but, for me,  the more we can get done over the summer (before teaching restarts), the better.

**Comment by nhorman** [comment]
So we're still in the planning stages here, we're not even sure if this is something we can commit to yet, though we really appreciate your willingness to work on this.  The current plan is:
1) Determine what the core developer team commitment to this might be
2) Document that and get approval from openssl leadership
3) Mark it as planned/todo or rejected on our planning board

That all said, you are correct in that the technical process is going be the creation of a feature branch, for which you can open an issue to do so, which will then get considered and acted upon, at which time you would be free to iterate on it, and that can be done independently of any planning we have to do here.

**Comment by mattcaswell** [comment]
To create the feature branch there is a process:

https://www.openssl.org/policies/general/feature-branch-approval-policy.html

> Submitting a request for a new feature branch should be done by creating an issue in the [openssl/project](https://github.com/openssl/project) repository. Such a request can be created by anyone. The request should specify:
> 
> - The name of the feature branch requested in the form of feature/<name>
> - A link to an issue describing the requested feature that will be developed on the branch
> - An estimate of when the work on the feature is expected to be complete
> - Whether there are any known resources being made available to work on the feature

Once approved creating the branch is a simple matter.

@sftcd - I'd suggest you are best place to create that issue

**Comment by sftcd** [comment]
Thanks both. Will do that feature branch request later today.

**Comment by sftcd** [comment]
Thanks to whomever decided for creating the feature branch! I made an initial PR for that that just adds the doc/designs/ech-api.md file. I'm sure someone will tell me if I did the wrong github things, which is always possible;-) That's https://github.com/openssl/openssl/pull/24738.

Also - is the plan to keep this issue open as a place to sync up and e.g. discuss what next to do with the feature branch etc.?

**Comment by nhorman** [comment]
Yes, I'm not aware of any other issue open for ECH, so we can track the effort here
